app/screens/browse_channels/browse_channels.tsx

```diff
@@ -0,0 +1,436 @@
+                        fetchChannels(
+                            serverUrl,
+                            currentTeamId,
+                            publicPage.current + 1,
+                            General.CHANNELS_CHUNK_SIZE,
+                        ).then(({channels: receivedChannels}) => loadedChannels(receivedChannels || [])); // Handle error?
+                    }
+                    break;
+                case SHARED:
+                    if (nextShared.current) {
+                        setLoading(true);
+                        fetchSharedChannels(
+                            serverUrl,
+                            currentTeamId,
+                            sharedPage.current + 1,
+                            General.CHANNELS_CHUNK_SIZE,
+                        ).then(({channels: receivedChannels}) => loadedChannels(receivedChannels || []));
+                    }
+                    break;
+                case ARCHIVED:
+                default:
+                    if (canShowArchivedChannels && nextArchived.current) {
+                        setLoading(true);
+                        fetchArchivedChannels(
+                            serverUrl,
+                            currentTeamId,
+                            archivedPage.current + 1,
+                            General.CHANNELS_CHUNK_SIZE,
+                        ).then(({channels: receivedChannels}) => loadedChannels(receivedChannels || []));
+                    }
+                    break;
+            }
+        }
+    }, [loading, typeOfChannels, joinedChannels, archivedChannels, channels, sharedChannels, serverUrl, currentTeamId]);
+
+    const selectActiveChannels = useCallback(() => {
+        switch (typeOfChannels) {
+            case ARCHIVED:
+                return archivedChannels;
+            case SHARED:
+                return sharedChannels;
+            default:
+                return channels;
+        }
+    }, [archivedChannels, channels, sharedChannels, typeOfChannels]);
+
+    const stopSearch = useCallback(() => {
+        setVisibleChannels(selectActiveChannels());
+        setTerm('');
+    }, [selectActiveChannels]);
+
+    const searchChannels = useCallback((text: string) => {
+        if (text) {
+            const active = selectActiveChannels();
+            const filtered = filterChannelsByTerm(active, text);
+            setTerm(text);
+            if (
+                filtered.length !== visibleChannels.length ||
+                filtered.reduce((shouldUpdate, c, i) => shouldUpdate || c.id !== visibleChannels[i].id, false)
+            ) {
+                setVisibleChannels(filtered);
+            }
+        } else {
+            stopSearch();
+        }
+    }, [selectActiveChannels, visibleChannels, stopSearch]);
+
+    const onPressDropdownElement = useCallback((channelType: string) => {
+        setTypeOfChannels(channelType);
+    }, []);
+
+    const renderLoading = useCallback(() => {
```

Comment:
this does not need a callback or to be memoized in my opinion, in fact you can do later on in this file

` ``
component = (
    <Loading
        containerStyle={style.loadingContainer}
...
```

Reply:
You are right. This was done like this when we were using the Loading component in two different places.